engine/lib/users.php

```diff
@@ -449,10 +449,16 @@ function user_avatar_hook($hook, $entity_type, $returnvalue, $params) {
 	$size = $params['size'];
 
 	if (isset($user->icontime)) {
-		return "avatar/view/$user->username/$size/$user->icontime";
```

Comment:
Never mind, I see that we have to take into account the 'view' and 'view_vars' hooks...